coil-core/src/androidMain/kotlin/coil3/RealImageLoader.kt

```diff
@@ -80,6 +82,15 @@ internal actual fun ComponentRegistry.Builder.addAndroidComponents(
         .add(DrawableFetcher.Factory())
         .add(BitmapFetcher.Factory())
         // Decoders
+        .apply {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
```

Comment:
Let's make this `if/else` so `BitmapFactoryDecoder.Factory` isn't added on API 28 and up.

Reply:
Same for this https://github.com/coil-kt/coil/pull/1969#discussion_r1436207509

Reply:
And I wonder if we should make ImageDecoder used only on API 29 and above, as ImageDecoder cannot safely decode a hardware bitmap on API 28, must use software allocation

Reply:
@VDostoyevskiy Yep, let's make it 29 and up then

Reply:
Done